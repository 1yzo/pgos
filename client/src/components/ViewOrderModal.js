import React from 'react';
import ReactModal from 'react-modal';
import moment from 'moment';
import '../styles/order-view.css';

class ViewOrderModal extends React.Component {
    render() {
        const { order } = this.props; 
        return (
            <ReactModal
                className="order-view"
                isOpen={this.props.showModal}
            >
                <div className="exit" onClick={() => this.props.closeModal()}>X</div>
                <div className="order-view__section">
                    <div className="order-view__label">Coffee</div>
                    <div className="order-view__info">{order.name}</div>
                </div>
                <div className="order-view__section">
                    <div className="order-view__label">Brew Method</div>
                    <div className="order-view__info">{order.brewMethod}</div>
                </div>
                <div className="order-view__section">
                    <div className="order-view__label">Ship Date</div>
                    <div className="order-view__info">{moment(order.shipDate).format('MM/DD/YYYY')}</div>
                </div>
                <div className="order-view__section">
                    <div className="order-view__label">Number of Cases</div>
                    <div className="order-view__info">{order.cases}</div>
                </div>
                <div className="order-view__section">
                    <div className="order-view__label">Packets per Case</div>
                    <div className="order-view__info">{order.packetsPerCase}</div>
                </div>
                {order.notes &&
                <div className="order-view__section">
                    <div className="order-view__label order-view__label--notes">Notes</div>
                    <div className="order-view__info">{order.notes}</div>
                </div>}
            </ReactModal>
        );
    }
}

export default ViewOrderModal;