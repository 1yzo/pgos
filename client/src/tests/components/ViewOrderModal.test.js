import React from 'react';
import { shallow } from 'enzyme';
import ViewOrderModal from '../../components/ViewOrderModal';
import ordersFixture from '../fixtures/orders';

test('should render ViewOrderModal correctly', () => {
    const wrapper = shallow(<ViewOrderModal order={ordersFixture[0]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should close modal on exit click', () => {
    const closeModal = jest.fn();
    const wrapper = shallow(<ViewOrderModal order={ordersFixture[0]} closeModal={closeModal} />);
    wrapper.find('.exit').simulate('click');
    expect(closeModal).toHaveBeenCalled();
})