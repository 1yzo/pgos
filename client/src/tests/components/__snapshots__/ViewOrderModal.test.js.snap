// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ViewOrderModal correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewOrderModal
    order={
        Object {
            "brewMethod": "Cold Brew",
            "cases": 5,
            "name": "Giant Steps",
            "notes": "Test notes",
            "packetsPerCase": 25,
            "priority": true,
            "shipDate": 10,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ariaHideApp": false,
      "bodyOpenClassName": "ReactModal__Body--open",
      "children": Array [
        <div
          className="exit"
          onClick={[Function]}
>
          X
</div>,
        <div
          className="order-view__section"
>
          <div
                    className="order-view__label"
          >
                    Coffee
          </div>
          <div
                    className="order-view__info"
          >
                    Giant Steps
          </div>
</div>,
        <div
          className="order-view__section"
>
          <div
                    className="order-view__label"
          >
                    Brew Method
          </div>
          <div
                    className="order-view__info"
          >
                    Cold Brew
          </div>
</div>,
        <div
          className="order-view__section"
>
          <div
                    className="order-view__label"
          >
                    Ship Date
          </div>
          <div
                    className="order-view__info"
          >
                    12/31/1969
          </div>
</div>,
        <div
          className="order-view__section"
>
          <div
                    className="order-view__label"
          >
                    Number of Cases
          </div>
          <div
                    className="order-view__info"
          >
                    5
          </div>
</div>,
        <div
          className="order-view__section"
>
          <div
                    className="order-view__label"
          >
                    Packets per Case
          </div>
          <div
                    className="order-view__info"
          >
                    25
          </div>
</div>,
        <div
          className="order-view__section"
>
          <div
                    className="order-view__label order-view__label--notes"
          >
                    Notes
          </div>
          <div
                    className="order-view__info"
          >
                    Test notes
          </div>
</div>,
      ],
      "className": "order-view",
      "closeTimeoutMS": 0,
      "isOpen": false,
      "parentSelector": [Function],
      "portalClassName": "ReactModalPortal",
      "shouldCloseOnEsc": true,
      "shouldCloseOnOverlayClick": true,
      "shouldFocusAfterRender": true,
      "shouldReturnFocusAfterClose": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X",
          "className": "exit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="order-view__label"
>
              Coffee
</div>,
            <div
              className="order-view__info"
>
              Giant Steps
</div>,
          ],
          "className": "order-view__section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Coffee",
              "className": "order-view__label",
            },
            "ref": null,
            "rendered": "Coffee",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Giant Steps",
              "className": "order-view__info",
            },
            "ref": null,
            "rendered": "Giant Steps",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="order-view__label"
>
              Brew Method
</div>,
            <div
              className="order-view__info"
>
              Cold Brew
</div>,
          ],
          "className": "order-view__section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Brew Method",
              "className": "order-view__label",
            },
            "ref": null,
            "rendered": "Brew Method",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cold Brew",
              "className": "order-view__info",
            },
            "ref": null,
            "rendered": "Cold Brew",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="order-view__label"
>
              Ship Date
</div>,
            <div
              className="order-view__info"
>
              12/31/1969
</div>,
          ],
          "className": "order-view__section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ship Date",
              "className": "order-view__label",
            },
            "ref": null,
            "rendered": "Ship Date",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "12/31/1969",
              "className": "order-view__info",
            },
            "ref": null,
            "rendered": "12/31/1969",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="order-view__label"
>
              Number of Cases
</div>,
            <div
              className="order-view__info"
>
              5
</div>,
          ],
          "className": "order-view__section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Number of Cases",
              "className": "order-view__label",
            },
            "ref": null,
            "rendered": "Number of Cases",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 5,
              "className": "order-view__info",
            },
            "ref": null,
            "rendered": 5,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="order-view__label"
>
              Packets per Case
</div>,
            <div
              className="order-view__info"
>
              25
</div>,
          ],
          "className": "order-view__section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Packets per Case",
              "className": "order-view__label",
            },
            "ref": null,
            "rendered": "Packets per Case",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 25,
              "className": "order-view__info",
            },
            "ref": null,
            "rendered": 25,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="order-view__label order-view__label--notes"
>
              Notes
</div>,
            <div
              className="order-view__info"
>
              Test notes
</div>,
          ],
          "className": "order-view__section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Notes",
              "className": "order-view__label order-view__label--notes",
            },
            "ref": null,
            "rendered": "Notes",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test notes",
              "className": "order-view__info",
            },
            "ref": null,
            "rendered": "Test notes",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ariaHideApp": false,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": Array [
          <div
            className="exit"
            onClick={[Function]}
>
            X
</div>,
          <div
            className="order-view__section"
>
            <div
                        className="order-view__label"
            >
                        Coffee
            </div>
            <div
                        className="order-view__info"
            >
                        Giant Steps
            </div>
</div>,
          <div
            className="order-view__section"
>
            <div
                        className="order-view__label"
            >
                        Brew Method
            </div>
            <div
                        className="order-view__info"
            >
                        Cold Brew
            </div>
</div>,
          <div
            className="order-view__section"
>
            <div
                        className="order-view__label"
            >
                        Ship Date
            </div>
            <div
                        className="order-view__info"
            >
                        12/31/1969
            </div>
</div>,
          <div
            className="order-view__section"
>
            <div
                        className="order-view__label"
            >
                        Number of Cases
            </div>
            <div
                        className="order-view__info"
            >
                        5
            </div>
</div>,
          <div
            className="order-view__section"
>
            <div
                        className="order-view__label"
            >
                        Packets per Case
            </div>
            <div
                        className="order-view__info"
            >
                        25
            </div>
</div>,
          <div
            className="order-view__section"
>
            <div
                        className="order-view__label order-view__label--notes"
            >
                        Notes
            </div>
            <div
                        className="order-view__info"
            >
                        Test notes
            </div>
</div>,
        ],
        "className": "order-view",
        "closeTimeoutMS": 0,
        "isOpen": false,
        "parentSelector": [Function],
        "portalClassName": "ReactModalPortal",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "exit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="order-view__label"
>
                Coffee
</div>,
              <div
                className="order-view__info"
>
                Giant Steps
</div>,
            ],
            "className": "order-view__section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Coffee",
                "className": "order-view__label",
              },
              "ref": null,
              "rendered": "Coffee",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Giant Steps",
                "className": "order-view__info",
              },
              "ref": null,
              "rendered": "Giant Steps",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="order-view__label"
>
                Brew Method
</div>,
              <div
                className="order-view__info"
>
                Cold Brew
</div>,
            ],
            "className": "order-view__section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Brew Method",
                "className": "order-view__label",
              },
              "ref": null,
              "rendered": "Brew Method",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cold Brew",
                "className": "order-view__info",
              },
              "ref": null,
              "rendered": "Cold Brew",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="order-view__label"
>
                Ship Date
</div>,
              <div
                className="order-view__info"
>
                12/31/1969
</div>,
            ],
            "className": "order-view__section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ship Date",
                "className": "order-view__label",
              },
              "ref": null,
              "rendered": "Ship Date",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "12/31/1969",
                "className": "order-view__info",
              },
              "ref": null,
              "rendered": "12/31/1969",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="order-view__label"
>
                Number of Cases
</div>,
              <div
                className="order-view__info"
>
                5
</div>,
            ],
            "className": "order-view__section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Number of Cases",
                "className": "order-view__label",
              },
              "ref": null,
              "rendered": "Number of Cases",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 5,
                "className": "order-view__info",
              },
              "ref": null,
              "rendered": 5,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="order-view__label"
>
                Packets per Case
</div>,
              <div
                className="order-view__info"
>
                25
</div>,
            ],
            "className": "order-view__section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Packets per Case",
                "className": "order-view__label",
              },
              "ref": null,
              "rendered": "Packets per Case",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 25,
                "className": "order-view__info",
              },
              "ref": null,
              "rendered": 25,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="order-view__label order-view__label--notes"
>
                Notes
</div>,
              <div
                className="order-view__info"
>
                Test notes
</div>,
            ],
            "className": "order-view__section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Notes",
                "className": "order-view__label order-view__label--notes",
              },
              "ref": null,
              "rendered": "Notes",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test notes",
                "className": "order-view__info",
              },
              "ref": null,
              "rendered": "Test notes",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
